modules/php/Cards/Rules/RulePoorBonus.php

```diff
@@ -16,14 +16,34 @@ public function __construct($cardId, $uniqueId)
     );
   }
 
-  public function immediateEffectOnPlay($player)
+  public function immediateEffectOnPlay($player_id)
   {
-    // @TODO : set game state?
-    // + draw extra card if current player is Poor
+    Utils::getGame()->setGameStateValue("activePoorBonus", 1);
+    if (Utils::hasLeastKeepers($player_id))
+    {
+      $addInflation = Utils::getActiveInflation() ? 1 : 0;
+
+      $poorBonus = 1 + $addInflation;
+      RulePoorBonus::notifyActiveFor($player_id);
+      Utils::getGame()->performDrawCards($player_id, $poorBonus);
```

Comment:
I think we need to update the game state with the number of drawn cards here. Because otherwise someone could play this card, get the extra card. Zap the rule to take it back and play it again and would get 2 extra cards. But the rule is "during their turn", so only once.
Same for the rich bonus.

Reply:
See my comment above, there is indeed more complications to this.
Maybe I would suggest to merge & include this branch as it is, to have as much as possible in the alfa release.
And consider this as additional changes in a separate branch for Rich & Poor.